WooCommerce/Classes/POS/Presentation/TotalsView.swift

```diff
@@ -33,31 +34,24 @@ struct TotalsView: View {
                             .transition(.opacity)
                     }
 
-                    if totalsViewModel.isShowingTotalsFields {
+                    if isShowingTotalsFields {
                         totalsFieldsView
                             .transition(.opacity)
                             .animation(.default, value: totalsViewModel.isShimmering)
+                            .opacity(totalsViewModel.isShowingTotalsFields ? 1 : 0)
                     }
                 }
                 .animation(.default, value: totalsViewModel.isShowingCardReaderStatus)
-                .transaction { transaction in
-                    // Disable animations within the view while the view is appearing
-                    if !hasViewAppeared {
-                        transaction.animation = nil
-                    }
-                }
```

Comment:
Perhaps instead of this fix, we can use a spacer where the card payment view will come in, so that the views below it are pushed down already?

Going back and forwards without changing the cart has actually got a bit worse than last I reviewed it with this: the spinner appears in front of the cart while it's transitioning to the left, when you go back to checkout.

Reply:
> Perhaps instead of this fix, we can use a spacer where the card payment view will come in, so that the views below it are pushed down already?

Not 100% sure if I understand what you mean, always hard to describe animations. 

If we put a spacer there and click check out for the first time, then total text fields will be pushed down even though they should be in the center. 

I couldn't figure out a reliable way to fix this yesterday. All the states are controlled with a view model. When payment is canceled, `card payment view` is hidden. When we enter the view after no changes, we skip order syncing and immediately call the service that starts to validate the order and ask for `card payment view` to appear. When we combine it with totalsView appearance transition, it seems that `card payment view` is appearing diagonally. 🤔 

> Going back and forwards without changing the cart has actually got a bit worse than last I reviewed it with this: the spinner appears in front of the cart while it's transitioning to the left, when you go back to checkout.

Yes, unfortunately when I connect the workaround with all the other animations in different PRs they interact in unexpected ways.